client/my-sites/sharing/buttons/appearance.jsx

```diff
@@ -1,115 +1,169 @@
-					showReblog={ ! this.props.site.jetpack && ( '' === this.props.values.disabled_reblogs || false === this.props.values.disabled_reblogs ) }
-					onLabelChange={ this.props.onChange.bind( null, 'sharing_label' ) }
+					showLike={
+						( ! this.props.isJetpack || this.props.isLikesModuleActive ) &&
+						( '' === this.props.values.disabled_likes || false === this.props.values.disabled_likes )
+					}
+					showReblog={
+						! this.props.isJetpack &&
+						( '' === this.props.values.disabled_reblogs || false === this.props.values.disabled_reblogs )
+					}
+					onLabelChange={ changeLabel }
 					onButtonsChange={ this.props.onButtonsChange } />
 			);
-		} else {
-			return <ButtonsPreviewPlaceholder />;
 		}
-	},
 
-	getReblogOptionElement: function() {
-		if ( ! this.props.site.jetpack ) {
+		return <ButtonsPreviewPlaceholder />;
+	}
+
+	getReblogOptionElement() {
+		if ( ! this.props.isJetpack ) {
 			return (
 				<label>
-					<input name="disabled_reblogs" type="checkbox" checked={ '' === this.props.values.disabled_reblogs || false === this.props.values.disabled_reblogs } onChange={ this.onReblogsLikesCheckboxClicked } disabled={ ! this.props.initialized } />
-					<span>{ this.translate( 'Show reblog button', { context: 'Sharing options: Checkbox label' } ) }</span>
+					<input
+						name="disabled_reblogs"
+						type="checkbox"
+						checked={ '' === this.props.values.disabled_reblogs || false === this.props.values.disabled_reblogs }
+						onChange={ this.onReblogsLikesCheckboxClicked }
+						disabled={ ! this.props.initialized }
+					/>
+					<span>{ this.props.translate( 'Show reblog button', { context: 'Sharing options: Checkbox label' } ) }</span>
 				</label>
 			);
 		}
-	},
+	}
 
-	getReblogLikeOptionsElement: function() {
-		if ( ! this.props.site.jetpack || this.props.site.isModuleActive( 'likes' ) ) {
+	getReblogLikeOptionsElement() {
+		if ( ( ! this.props.isJetpack || this.props.isLikesModuleActive ) ) {
 			return (
 				<fieldset className="sharing-buttons__fieldset">
 					<legend className="sharing-buttons__fieldset-heading">
-						{ this.translate( 'Reblog & Like', { context: 'Sharing options: Header' } ) }
+						{ this.props.translate( 'Reblog & Like', { context: 'Sharing options: Header' } ) }
 					</legend>
 					{ this.getReblogOptionElement() }
 					<label>
-						<input name="disabled_likes" type="checkbox" checked={ '' === this.props.values.disabled_likes || false === this.props.values.disabled_likes } onChange={ this.onReblogsLikesCheckboxClicked } disabled={ ! this.props.initialized } />
-						<span>{ this.translate( 'Show like button', { context: 'Sharing options: Checkbox label' } ) }</span>
+						<input
+							name="disabled_likes"
+							type="checkbox"
+							checked={ '' === this.props.values.disabled_likes || false === this.props.values.disabled_likes }
```

Comment:
I think it's not possible to use a selector here, since it's computed from the parent's component state, but I factorized it in a small class method anyway.